diff --git a/node_modules/node-zklib/zklibtcp.js b/node_modules/node-zklib/zklibtcp.js
index 194a9d8..65324b2 100644
--- a/node_modules/node-zklib/zklibtcp.js
+++ b/node_modules/node-zklib/zklibtcp.js
@@ -100,7 +100,7 @@ class ZKLibTCP {
           timer = await setTimeout(() => {
             clearTimeout(timer)
             reject(new Error('TIMEOUT_ON_WRITING_MESSAGE'))
-          }, connect ? 2000 : this.timeout)
+          }, this.timeout)
         }
       })
     })
@@ -229,6 +229,7 @@ class ZKLibTCP {

       } catch (err) {
         reject(err)
+        return
       }

       const header = decodeTCPHeader(reply.subarray(0, 16))
